wazo_call_logd/database/models.py

```diff
@@ -108,6 +116,42 @@ class CallLog(Base):
     )
 
 
+@generic_repr
+class Destination(Base):
+
+    __tablename__ = 'call_logd_call_log_destination'
+
+    uuid = Column(
+        UUIDType,
+        server_default=text('uuid_generate_v4()'),
+        primary_key=True,
+    )
+
+    call_log_id = Column(
+        Integer,
+        ForeignKey(
+            'call_logd_call_log.id',
+            name='call_logd_call_log_destination_call_log_id_fkey',
+            ondelete='CASCADE',
+        ),
+    )
+
+    type = Column(String(10), nullable=False)
+    conference_id = Column(String(10))
+    user_uuid = Column(String(38))
+    meeting_uuid = Column(String(38))
+    user_name = Column(String(255))
+    meeting_name = Column(String(255))
```

Comment:
That's not the approved design in the ticket
If you don't know the difference, please ask and stop trying to silence design modification

Reply:
I followed the following suggestion in the ticket:
```
destination_details: {  # new API field
     type: user/conference/meeting/unknown,
     user_uuid: xxx,
     conference_id: xxx,
     meeting_uuid: xxx,
   }
```
What do you mean by that? can you elaborate please? Also, in the database design proposition in the ticket:

```
destination_details_key: string. Conventional values: type, user_uuid, conference_id, meeting_uuid. Can be added as DB constraint, but not an enum.
destination_details_value: string.
```
the `destination_details_value` cannot be a string according to the discussion in the thread; because we agreed to include `meeting_name` with `meeting_uuid` for meetings, and `user_name` with `user_uuid` for users (as for conferences, we will use only `conference_id`). So that is why I included `meeting_name` and `user_name` fields. Also, I did not use `hstore` type because I was asked to remove it.

Reply:
Yes I talked about the `destination_details_key` and `destination_details_value`
I don't understand why you cannot store integer into a string and cast the value before exposing the value?
call-logs are not intended to make relationship on the DB side, it's purely to logs information at time X

Also storing a row with `type: null` is unacceptable. This table can have millions of entries and you don't want to create empty row if not necessary

Reply:
Yes, that is correct. So the `destination` table should only have 3 fields: `call_log_id`, `destination_details_key` and `destination_details_value`, as described in the [ticket specs](https://wazo-dev.atlassian.net/browse/WAZO-2826). You will probably need to keep the `uuid` column for SQLAlchemy to be happy, so that would make 4 columns. I'll add that to the ticket specs.

> you want to remove the relationship

There is no need to remove any relationship, though, so I don't understand this part.

Reply:
> Yes, that is correct. So the `destination` table should only have 3 fields: `call_log_id`, `destination_details_key` and `destination_details_value`, as described in the [ticket specs](https://wazo-dev.atlassian.net/browse/WAZO-2826). You will probably need to keep the `uuid` column for SQLAlchemy to be happy, so that would make 4 columns. I'll add that to the ticket specs.
> 
> > you want to remove the relationship
> 
> There is no need to remove any relationship, though, so I don't understand this part.

Thanks Sebastien, Just one question: since Francois wants to remove the default `unknown` value; what should it be by default?

Reply:
> Just one question: since Francois wants to remove the default `unknown` value; what should it be by default?

The API should return `unknown` as a default value for `destination_details.type`. But it should not be stored in the database. If there is no entry in the `destination` table linked to a `call_log_id`, then the type is considered as `unknown`.

Reply:
I have added this remark in the [ticket specs](https://wazo-dev.atlassian.net/browse/WAZO-2826)